code/modules/ntsl2/ntsl2.dm

```diff
@@ -0,0 +1,46 @@
+/NTSL_interpreter
+	var/connected = 0
+	var/list/programs = list()
+
+	New()
+		spawn(300)
+			attempt_connect()
```

Comment:
```suggestion
		addtimer(CALLBACK(src, .proc/attempt_connect), 30 SECONDS)
```
No sleeps, use timers and callbacks instead. Why the 30 second delay?

Reply:
Sleep is okay, spawn is not.

Reply:
Timer was removed as it was used for testing purposes.